josh-core/src/history.rs

```diff
@@ -256,9 +272,12 @@ fn find_new_branch_base(
     };
     tracing::info!("new branch base?");
 
+    // Walk filtered history tree, trying to find a base for every commit
```

Comment:
remove the word `tree` -> It has different meaning in git that is misleading here. The sentence works without it.